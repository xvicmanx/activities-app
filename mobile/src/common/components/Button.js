import React from 'react';
import { View, StyleSheet } from 'react-native';
import { Button as ButtonRNE } from 'react-native-elements';
import { COLORS } from '../../constants';

const Button = ({ style, icon, danger, small, children, loading, ...rest }) => {
  const smallStyles = small && styles.small;
  const dangerStyles = danger && styles.danger;

  return (
    <ButtonRNE
      title={children}
      loading={loading}
      buttonStyle={{
        backgroundColor: COLORS.primary,
        height: 50,
        ...smallStyles,
        ...dangerStyles,
        ...style,
      }}
      icon={icon && <View style={styles.iconContainer}>{icon}</View>}
      iconRight
      {...rest}
    />
  );
};

const styles = StyleSheet.create({
  small: {
    alignSelf: 'flex-start',
    paddingHorizontal: 20,
    height: 35,
  },
  danger: {
    backgroundColor: COLORS.danger,
  },
  iconContainer: {
    marginLeft: 10,
  },
});

export default Button;
