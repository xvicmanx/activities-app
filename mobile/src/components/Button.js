import React from 'react';
import { View } from 'react-native';
import { Button as ButtonRNE } from 'react-native-elements';
import { COLORS } from '../constants/colors';

const Button = ({ icon, danger, small, children, loading, ...rest }) => {
  const smallStyles = small && {
    alignSelf: 'flex-start',
    paddingHorizontal: 20,
    height: 35,
  };

  const dangerStyles = danger && {
    backgroundColor: '#e74c3c',
  };

  return (
    <ButtonRNE
      title={children}
      loading={loading}
      buttonStyle={{
        backgroundColor: COLORS.primary,
        height: 50,
        ...smallStyles,
        ...dangerStyles,
      }}
      icon={icon && <View style={{ marginLeft: 10 }}>{icon}</View>}
      iconRight
      {...rest}
    />
  );
};

export default Button;
