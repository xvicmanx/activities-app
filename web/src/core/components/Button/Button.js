import React, { useState } from 'react';
import { Button as BulmaButton } from 'react-bulma-components';

const Button = ({ children, className, hoverClassName, ...rest }) => {
  const [hovered, setHovered] = useState(false);
  const classes = [];

  if (className) {
    classes.push(className);
  }

  if (hovered && hoverClassName) {
    classes.push(hoverClassName);
  }

  return (
    <BulmaButton
      {...rest}
      onMouseEnter={() => {
        setHovered(true);
      }}
      className={classes.join(' ')}
    >
      {children}
    </BulmaButton>
  );
};

Button.Group = BulmaButton.Group;

export default Button;
