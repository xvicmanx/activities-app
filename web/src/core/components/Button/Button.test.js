import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Button from './Button';

describe('Button', () => {
  it('renders properly', () => {
    const renderer = new ShallowRenderer();
    const result = renderer.render(<Button className="test">Hello</Button>);
    expect(result).toMatchSnapshot();
  });
});
