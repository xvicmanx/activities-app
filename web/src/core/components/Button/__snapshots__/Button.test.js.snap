// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders properly 1`] = `
<Button
  className="test"
  disabled={false}
  fullwidth={false}
  inverted={false}
  isSelected={false}
  isStatic={false}
  loading={false}
  onClick={[Function]}
  outlined={false}
  remove={false}
  renderAs="button"
  reset={false}
  rounded={false}
  submit={false}
  text={false}
>
  Hello
</Button>
`;
